src/trg-main-window.c

```diff
@@ -1514,9 +1514,15 @@ trg_torrent_tree_view_visible_func(GtkTreeModel * model,
             gchar *filterCmp = g_utf8_casefold(filterText, -1);
             gchar *nameCmp = g_utf8_casefold(name, -1);
 
-            if (!strstr(nameCmp, filterCmp))
-                visible = FALSE;
+            GRegex *regex;
+            GMatchInfo *match_info;
+            regex = g_regex_new (filterCmp, 0, 0, NULL);
 
+            if (!g_regex_match (regex, nameCmp, 0, &match_info))
+	        visible = FALSE;
```

Comment:
This logic also assumes its always regex which just isn't true and a string that is valid regex may also be a literal somebody wanted to search for.

I think this needs some UI work honestly to just have the user toggle regex or not.

Reply:
it's easy enough to escape regex sequences; the only "common" character usually searched for that's also a special char in regex is `.` , which you can just escape with `\.` (sure also brackets and parentesis could be there, but i would highly unlikely someone would actually search for them: you likely start typing a string not weird special characters).

I'm afraid i will not be able to contribute any UI change.

Reply:
Regex has plenty of special characters... `()[]+.\`, etc. Its silly to expect users to understand that.

Anyway I'll just close this PR if you won't be doing more in-depth work.